src/js/components/sharedComponents/dropdowns/NewPicker.jsx

```diff
@@ -0,0 +1,174 @@
+        };
+    }, [expanded, id]);
+
+    const toggleMenu = (e) => {
+        e.preventDefault();
+        setExpanded(!expanded);
+    };
+
+    const keyUp = (e) => {
+        if (e.key === "Escape" && expanded) {
+            setExpanded(!expanded);
+        }
+    };
+
+    const handleSort = (a, b) => sortFn(a, b, selectedOption);
+    // may need this when going from disabled to enabled in the future
+    // const enableButton = (e) => {
+    //     e.preventDefault();
+    //     if (!isEnabled) {
+    //         setIsEnabled(!isEnabled);
+    //     }
+    // };
+    const createOnClickFn = (cb) => (param) => {
+        cb(param);
+        setExpanded(false);
+    };
+
+    let variation = '';
+    if (size === 'sm' || size === 'small') {
+        variation = '-sm';
+    }
+    else if (size === 'md' || size === 'medium') {
+        variation = '-md';
+    }
+    else if (size === 'lg' || size === 'large') {
+        variation = '-lg';
+    }
+
+    return (
+        <div className="filter__dropdown-container" ref={pickerRef}>
+            {label !== '' && <span className={`filter__dropdown-label${variation} ${isEnabled ? 'enabled' : 'not-enabled'}`}>{label}</span>}
+            <div className="filter__dropdown-button-list-container">
+                <button
+                    className={`filter__dropdown-button${variation} ${isEnabled ? 'enabled' : 'not-enabled'}`}
+                    ref={buttonRef}
+                    aria-label="Filter Dropdown Button"
+                    onClick={toggleMenu}
+                    onKeyUp={keyUp}>
+                    <span className="filter__dropdown-left-icon">
+                        <FontAwesomeIcon icon={leftIcon} alt="page title bar button icon" />
+                    </span>
+                    {children ?
+                        <>{children}</> :
+                        <span className="filter__dropdown-button-text">
+                            {selectedOption}
+                        </span>
+                    }
+                    <span className="filter__dropdown-chevron">
+                        {!expanded && (
+                            <FontAwesomeIcon icon="chevron-down" alt="Toggle menu" />
+                        )}
+                        {expanded && (
+                            <FontAwesomeIcon icon="chevron-up" alt="Toggle menu" />
+                        )}
+                    </span>
+                </button>
+                <ul className={`filter__dropdown-list${variation} ${expanded ? '' : 'hide'} ${isEnabled ? 'enabled' : 'not-enabled'} ${dropdownClassName}`}>
+                    {options?.sort(handleSort)
+                        .map((option) => ({
+                            ...option,
+                            onClick: createOnClickFn(option.onClick)
+                        }))
+                        .map((option) => {
+                            console.debug("option map: ", option);
```

Comment:
i missed this one the first time, thought there was only one

Reply:
bah i also missed it